### PR TITLE
Lints local imports. Closes #1197

### LINKED ISSUE #1197: Set-up opinionated code imports and formatting
To avoid having PRs with odd formatting, set-up an opinionated code formatter and linter

### PR DESCRIPTION
**Pull request by alexec**

### CODE REVIEW

File: gometalinter.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "Enable": [
         "vet",
         "gofmt",
+        "goimports",
```
Comment (alexec) [inline gometalinter.json]: Key change.

File: hack/format-code.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@ STAGED_GO_FILES=$(git diff --cached --name-only | grep ".go$" || true)
 
 if [[ "${STAGED_GO_FILES}" != "" ]]; then
     echo "Formatting imports"
-    goimports -w ${STAGED_GO_FILES} ;
+    goimports -w -local github.com/argoproj/argo-cd ${STAGED_GO_FILES} ;
```
Comment (alexec) [inline hack/format-code.sh]: Key change.

File: Makefile [hunk 3]
```diff
@@ -55,7 +55,7 @@ clientgen:
 	./hack/update-codegen.sh
 
 .PHONY: codegen
-codegen: protogen clientgen
+codegen: protogen clientgen format-code
```
Comment (alexec) [inline Makefile]: Key change.

File: hack/format-code.sh [hunk 4]
```diff
@@ -1,14 +1,14 @@
 #! /bin/sh
 set -eu
 
-STAGED_GO_FILES=$(git diff --cached --name-only | grep ".go$" || true)
```
Comment (alexec) [inline hack/format-code.sh]: Key change.

File: server/cluster/cluster.go [hunk 5]
```diff
@@ -12,6 +12,8 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
 
+	log "github.com/sirupsen/logrus"
+
```
Comment (jessesuen) [inline server/cluster/cluster.go]: Looks like this should be grouped with above

File: server/project/project.go [hunk 6]
```diff
@@ -12,6 +12,8 @@ import (
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/client-go/kubernetes"
 
+	jwt "github.com/dgrijalva/jwt-go"
```
Comment (jessesuen) [inline server/project/project.go]: Group with above

File: server/server.go [hunk 7]
```diff
@@ -37,6 +37,8 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
 
+	"github.com/improbable-eng/grpc-web/go/grpcweb"
```
Comment (jessesuen) [inline server/server.go]: Group with above

File: util/cli/cli.go [hunk 8]
```diff
@@ -16,13 +16,14 @@ import (
 
 	"k8s.io/kubernetes/pkg/kubectl/util/term"
```
Comment (jessesuen) [inline util/cli/cli.go]: Group with below

File: util/db/db_test.go [hunk 9]
```diff
@@ -15,6 +14,8 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/fake"
 
+	"github.com/argoproj/argo-cd/common"
```
Comment (jessesuen) [inline util/db/db_test.go]: group with below

**Review by jessesuen — APPROVED** [review]

**Pull request merged**